core/domain/suggestion_services.py

```diff
@@ -2967,3 +2968,196 @@ def enqueue_contributor_ranking_notification_email_task(
+    """Returns data to generate the certificate.
+
+    Args:
+        username: str. The username of the contributor.
+        language_code: str|None. The language for which the contributions should
+            be considered.
+        suggestion_type: str. The type of suggestion that the certificate
+            needs to generate.
+        from_date: datetime.datetime. The start of the date range for which the
+            contributions were created.
+        to_date: datetime.datetime. The end of the date range for which the
+            contributions were created.
+
+    Returns:
+        ContributorCertificateInfoDict. Data to generate the certificate.
+
+    Raises:
+        Exception. The suggestion type is invalid.
+        Exception. There is no user for the given username.
+    """
+    user_id = user_services.get_user_id_from_username(username)
+    if user_id is None:
+        raise Exception('There is no user for the given username.')
+
+    if suggestion_type == feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT:
+        data = _generate_translation_contributor_certificate_data(
+            language_code, from_date, to_date, user_id)
+
+    elif suggestion_type == feconf.SUGGESTION_TYPE_ADD_QUESTION:
+        data = _generate_question_contributor_certificate_data(
+            from_date, to_date, user_id)
+
+    else:
+        raise Exception('The suggestion type is invalid.')
+
+    return data.to_dict()
+
+
+def _generate_translation_contributor_certificate_data(
+    language_code: Optional[str],
+    from_date: datetime.datetime,
+    to_date: datetime.datetime,
+    user_id: str
+) -> suggestion_registry.ContributorCertificateInfo:
+    """Returns data to generate translation submitter certificate.
+
+    Args:
+        language_code: str|None. The language for which the contributions should
+            be considered.
+        from_date: datetime.datetime. The start of the date range for which
+            the contributions were created.
+        to_date: datetime.datetime. The end of the date range for which
+            the contributions were created.
+        user_id: str. The user ID of the contributor.
+
+    Returns:
+        ContributorCertificateInfo. Data to generate translation submitter
+        certificate.
+
+    Raises:
+        Exception. The language is invalid.
+    """
+    signature = feconf.TRANSLATION_TEAM_LEAD
+
+    # Adds one date to the to_date to make sure the contributions within
+    # the to_date are also counted for the certificate.
+    to_date_to_fetch_contributions = to_date + datetime.timedelta(days=1)
+
+    language = next(filter(
+        lambda lang: lang['id'] == language_code,
+        constants.SUPPORTED_AUDIO_LANGUAGES), None)
+    if language is None:
+        raise Exception('The provided language is invalid.')
+    # Since we have found a corresponding language for the given language code
+    # we can make sure that language code is not None.
```

Comment:
Wait why do we allow `None` to be passed as `language_code` when we know that it will fail?

Reply:
I think it would be better to handle this outside this function.

Reply:
Done